samples/dnn/speech_recognition.cpp

```diff
@@ -0,0 +1,581 @@
+        "{input_audio a      |                          | Path to input audio file        }"
+        "{audio_stream       | 0                        | CAP_PROP_AUDIO_STREAM value     }"
+        "{show_spectrogram s | off                      | Show a spectrogram of the input audio: on or off }"
+        "{model m            | jasper.onnx              | Path to the onnx file of Jasper }"
+        "{backend b          | dnn::DNN_BACKEND_DEFAULT | Select a computation backend: "
+                                                          "dnn::DNN_BACKEND_DEFAULT, "
+                                                          "dnn::DNN_BACKEND_INFERENCE_ENGINE, "
+                                                          "dnn::DNN_BACKEND_OPENCV }"
+        "{target t           | dnn::DNN_TARGET_CPU      | Select a target device: "
+                                                          "dnn::DNN_TARGET_CPU, "
+                                                          "dnn::DNN_TARGET_OPENCL, "
+                                                          "dnn::DNN_TARGET_OPENCL_FP16 }"
+        ;
+    CommandLineParser parser(argc, argv, keys);
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    vector<double>inputAudio = {};
+    int samplingRate = 0;
+    if (parser.get<std::string>("input_type") == "file")
+    {
+        string audio = samples::findFile(parser.get<std::string>("input_audio"));
+        samplingRate = readAudioFile(inputAudio, audio, parser.get<int>("audio_stream"));
+    }
+    else if (parser.get<std::string>("input_type") == "microphone")
+    {
+        samplingRate = readAudioMicrophone(inputAudio, parser.get<int>("micro_time"));
+    }
+
+    if ((inputAudio.size() == 0) || samplingRate <= 0)
+    {
+        cerr << "Error: problems with audio reading, check input arguments" << endl;
+        return -1;
+    }
+
+
+    FilterbankFeatures filter;
+    auto calculated_features = filter.calculate_features(inputAudio);
+
+    // Show spectogram if required
+    if (parser.get<std::string>("show_spectrogram") == "on")
+    {
+        Mat spectogram;
+        Mat featuresMat(static_cast<int>(calculated_features.size()), static_cast<int>(calculated_features[0].size()), CV_64F);
+        for(int i = 0; i < featuresMat.size[0]; ++i)
+        {
+            for(int j = 0; j < featuresMat.size[1]; ++j)
+            {
+                featuresMat.at<double>(i, j) = calculated_features[i][j];
+            }
+        }
+        normalize(featuresMat, spectogram, 0, 255, NORM_MINMAX, CV_8U);
+        applyColorMap(spectogram, spectogram, COLORMAP_INFERNO);
+        imshow("spectogram", spectogram);
+        waitKey(0);
+    }
+
+    // Load Network
+    dnn::Net net = dnn::readNetFromONNX(parser.get<std::string>("model"));
+    net.setPreferableBackend(parser.get<int>("backend"));
+    net.setPreferableTarget(parser.get<int>("target"));
```

Comment:
To improve user experience it makes sense to load network before capturing audio (to avoid surprises after capturing from microphone).